lib/openid_connect/authorization_point/endpoint_confirmation_point.rb

```diff
@@ -0,0 +1,34 @@
+module OpenidConnect
+  module AuthorizationPoint
+    class EndpointConfirmationPoint < Endpoint
+      def initialize(current_user, approved=false)
+        super(current_user)
+        @approved = approved
+      end
+
+      def handle_response_type(req, res)
+        handle_approval(@approved, req, res)
+      end
+
+      def handle_approval(approved, req, res)
+        if approved
+          approved!(req, res)
+        else
+          req.access_denied!
+        end
+      end
+
+      def approved!(req, res)
+        auth = OpenidConnect::Authorization.find_or_create(req.client_id, @user)
+        response_types = Array(req.response_type)
+        if response_types.include?(:id_token)
+          id_token = auth.id_tokens.create!(nonce: req.nonce)
+          options = %i(code access_token).map{|option| ["res.#{option}", res.respond_to?(option) ? res.option : nil]}.to_h
```

Comment:
Line is too long. [122/120]<br>Space missing to the left of {.<br>Space missing inside }.